pkg/load/agents/configAgent_test.go

```diff
@@ -0,0 +1,174 @@
+		},
+	}
+
+	wg := &sync.WaitGroup{}
+	for i := 0; i < 2; i++ {
+		wg.Add(2)
+
+		go func() { _, _ = agent.GetFromIndex("bla", "blub"); wg.Done() }()
+		go func() {
+			_ = agent.AddIndex("foo", func(_ api.ReleaseBuildConfiguration) []string {
+				return []string{"bar"}
+			})
+			wg.Done()
+		}()
+	}
+	wg.Wait()
+
+}
+
+func TestAddIndex(t *testing.T) {
+	agent := &configAgent{
+		lock: &sync.RWMutex{},
+		indexFuncs: map[string]IndexFn{
+			"exists": func(_ api.ReleaseBuildConfiguration) []string { return nil },
+		},
+	}
+	testCases := []struct {
+		name          string
+		indexFnName   string
+		expectedError string
+	}{
+		{
+			name:        "Happy path",
+			indexFnName: "new",
+		},
+		{
+			name:          "Index already exists",
+			indexFnName:   "exists",
+			expectedError: `there is already an index named "exists"`,
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			tc := tc
+			// Run in parallel so race detector can do its job
+			t.Parallel()
+
+			errMsg := ""
+			err := agent.AddIndex(tc.indexFnName, func(_ api.ReleaseBuildConfiguration) []string { return nil })
+			if err != nil {
+				errMsg = err.Error()
+			}
+
+			if errMsg != tc.expectedError {
+				t.Errorf("expected error %s, got error %s", tc.expectedError, errMsg)
+			}
+		})
+	}
+}
+
+func TestBuildIndexes(t *testing.T) {
+
+	cfg := api.ReleaseBuildConfiguration{TestBinaryBuildCommands: "make test"}
+	testCases := []struct {
+		name     string
+		agent    *configAgent
+		configs  FilenameToConfig
+		expected map[string]configIndex
+	}{
+		{
+			name: "single index",
+			agent: &configAgent{
+				indexFuncs: map[string]IndexFn{
+					"index-a": func(_ api.ReleaseBuildConfiguration) []string { return []string{"key-a"} },
+				},
+			},
+			configs:  FilenameToConfig{"myfile.yaml": cfg},
+			expected: map[string]configIndex{"index-a": {"key-a": []*api.ReleaseBuildConfiguration{&cfg}}},
+		},
+		{
+			name: "multiple indexes",
+			agent: &configAgent{
+				indexFuncs: map[string]IndexFn{
+					"index-a": func(_ api.ReleaseBuildConfiguration) []string { return []string{"key-a"} },
+					"index-b": func(_ api.ReleaseBuildConfiguration) []string { return []string{"key-b"} },
+				},
+			},
+			configs: FilenameToConfig{"myfile.yaml": cfg},
+			expected: map[string]configIndex{
+				"index-a": {"key-a": []*api.ReleaseBuildConfiguration{&cfg}},
+				"index-b": {"key-b": []*api.ReleaseBuildConfiguration{&cfg}},
+			},
+		},
+		{
+			name: "no result indexer",
+			agent: &configAgent{
+				indexFuncs: map[string]IndexFn{
+					"index-a": func(_ api.ReleaseBuildConfiguration) []string { return nil },
+				},
+			},
+			configs:  FilenameToConfig{"myfile.yaml": cfg},
+			expected: map[string]configIndex{},
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			tc.agent.indexes = tc.agent.buildIndexes(tc.configs)
+			if diff := deep.Equal(tc.agent.indexes, tc.expected); diff != nil {
```

Comment:
we usually use `cmp.Diff()` -- why is this better?

Reply:
Updated the pull to use `cmp.Diff`